### DISCUSSION TITLE
Restarting DFT calculation from DF electron density

### DISCUSSION THREAD

**Discussion by songfeitong** [pyscf/pyscf#2682] [Q&A]
Assume I have the electron density represented as the expansion coefficients under an DF auxiliary basis (which come from machine learning prediction). Although the kinetic energy cannot be evaluated without an AO basis, theoretically the Kohn-Sham effective potential has no kinetic term and should be able to be constructed from DF density (assume we are using PBE). Then we might restart the DFT from this effective potential.

Is this possible in pyscf? And if so, may I have some hints on how to do this?

**Comment by fishjojo** [comment]
The XC potential can be computed with https://github.com/pyscf/pyscf/blob/14d88828cd1f18f1e5358da1445355bde55322a1/pyscf/dft/numint.py#L2650 and https://github.com/pyscf/pyscf/blob/14d88828cd1f18f1e5358da1445355bde55322a1/pyscf/dft/numint.py#L557, where `rho` is the electron density on the grid. The Coulomb potential may be computed with 3 center 2 electron integrals, `mol.intor( 'int3c2e' )`.

**Comment by songfeitong** [comment]
@fishjojo Thanks for your reply! Since I am new to pyscf, may I ask for a bit more detailed instruction? Do I have to hack the source code to do this, or I can simply overwrite some functions?

**Reply by sunqm** [reply]
If you can transform the density into density matrix on AO bases, the density matrix can be passed to the RKS.kernel function as `mf.kernel(dm)`. Pyscf does not support directly reading a density as initial guess.
